### PR TITLE
Try out html2canvas library

### PR DESCRIPTION
**Pull request by li-kai**
Well just so happens they have a new rewrite that allows you to ignore elements without having to interfere with user \o/

Docs: http://html2canvas.hertzen.com/getting-started

### CODE REVIEW

File: v3/src/js/actions/export.js [hunk 1]
```diff
@@ -26,20 +26,19 @@ export function downloadAsJpeg(domElement: Element) {
...
-      })
-      .catch((err) => {
-        console.error(err); // eslint-disable-line
-        dispatch({
-          type: `${DOWNLOAD_AS_JPEG}_FAILURE`,
-        });
+    const config = { foreignObjectRendering: true, logging: process.env.NODE_ENV === 'development' };
+    return html2canvas(domElement, config).then((canvas) => {
+      const dataUrl = canvas.toDataURL('image/png');
+      downloadUrl(dataUrl, 'timetable.jpeg');
```
Comment (ZhangYiJiang) [inline v3/src/js/actions/export.js]: Should be `.png`, not JPEG

**Comment by yangshun** [comment]
Not sure why, but the exported image seems cut off

timetable 25

**Comment by yangshun** [comment]
This works on Safari! There's still the problem of timetable clipping and weird visual artifacts on the left but it's still better than the current version. Let's use yours instead over https://github.com/nusmodifications/nusmods/pull/554?

timetable-3

File: v3/src/js/actions/export.js [hunk 2]
```diff
@@ -26,20 +26,19 @@ export function downloadAsJpeg(domElement: Element) {
...
+      dispatch({
+        type: `${DOWNLOAD_AS_JPEG}_SUCCESS`,
```
Comment (yangshun) [inline v3/src/js/actions/export.js]: We should rename the action and type to `downloadAsImage`/`DOWNLOAD_AS_IMAGE` once and for all.

File: v3/src/js/actions/export.js [hunk 3]
```diff
@@ -26,20 +26,19 @@ export function downloadAsJpeg(domElement: Element) {
...
       });
+    }).catch((err) => {
+      console.error(err); // eslint-disable-line
```
Comment (yangshun) [inline v3/src/js/actions/export.js]: Probably log the error in Sentry here

**Comment by li-kai** [comment]
Closed in favor #596

**Pull request closed without merging**